File: ui/src/lib/models/repo.ts
export interface Repo {
	id: string;
	owner: string;
	name: string;
	full_name: string;
	created_at: string;
}

export interface GitHubRepo {
	full_name: string;
	owner_login: string;
	name: string;
	description: string;
	private: boolean;
	html_url: string;
}

File: internal/repo/repo.go
package repo

import (
	"fmt"
	"strings"
	"time"
)

// Repo represents a GitHub repository added to Verve.
type Repo struct {
	ID        RepoID    `json:"id"`
	Owner     string    `json:"owner"`
	Name      string    `json:"name"`
	FullName  string    `json:"full_name"`
	CreatedAt time.Time `json:"created_at"`
}

// NewRepo creates a new Repo from a full name (e.g., "owner/repo").
func NewRepo(fullName string) (*Repo, error) {
	parts := strings.SplitN(fullName, "/", 2)
	if len(parts) != 2 || parts[0] == "" || parts[1] == "" {
		return nil, fmt.Errorf("invalid repo full name %q: expected owner/name", fullName)
	}
	return &Repo{
		ID:        NewRepoID(),
		Owner:     parts[0],
		Name:      parts[1],
		FullName:  fullName,
		CreatedAt: time.Now(),
	}, nil
}
